ocamlrep/impls.rs

```diff
@@ -770,6 +770,11 @@ impl ToOcamlRep for OsStr {
         use std::os::unix::ffi::OsStrExt;
         alloc.add(self.as_bytes())
     }
+
+    #[cfg(target_arch = "wasm32")]
+    fn to_ocamlrep<'a, A: Allocator>(&'a self, alloc: &'a A) -> Value<'a> {
+        panic!()
```

Comment:
I'd like to avoid having `panic!` like this.
As @jakeblakeley  points out:
> FWIW looks like wasm OsStrs are represented the same way as unix ones: https://github.com/rust-lang/rust/blob/master/library/std/src/os/wasi/ffi.rs#L7
Oh, looks like implementing this is easy, actually: I think they just need to write the same thing as the unix implementation but import [std::os::wasi::ffi::OsStrExt](https://doc.rust-lang.org/std/os/wasi/ffi/trait.OsStrExt.html) instead of the unix version.

Would you consider doing it in this PR, or do you want to wait for a subsequent one ?

Reply:
Incredibly happy to make the change in this PR! Will update today

Reply:
> As @jakeblakeley points out:

you meant @rjbailey here 😀

Reply:
oups, right.. autocompletion... sorry !

Reply:
I tried using [std::os::wasi::ffi::OsStrExt](https://doc.rust-lang.org/std/os/wasi/ffi/trait.OsStrExt.html) but it failed when compiling with the target `wasm32-unknown-unknown`:

```
776 |         use std::os::wasi::ffi::OsStrExt;
    |                      ^^^^ could not find `wasi` in `os`
```

I believe this is by design, and it makes sense — a WASM container running in a browser doesn't have any concept of FFI.